pkg/alertmanager/multitenant.go

```diff
@@ -314,12 +321,13 @@ func (am *MultitenantAlertmanager) syncConfigs(cfgs map[string]alerts.AlertConfi
 	for user, cfg := range cfgs {
 		err := am.setConfig(cfg)
 		if err != nil {
-			am.multitenantMetrics.invalidConfig.WithLabelValues(user).Set(float64(1))
+			am.multitenantMetrics.lastReloadSuccessful.WithLabelValues(user).Set(float64(0))
+			am.multitenantMetrics.lastReloadSuccessfulTimestamp.WithLabelValues(user).SetToCurrentTime()
```

Comment:
If there's an error here that means we didn't reload successfully, shouldn't this go after L330?

Reply:
Oh yes, this is not correct. Good job @gotjosh spotting this issue!

Reply:
gah oops 🤦  1816fe864
would also appreciate any pointers on how to write a proper unit test for this timestamp metric, at first glance i couldn't find any examples to follow, and i'd rather not go the approach of monkey patching `time.Now()` 😬